.circleci/config.yml

```diff
@@ -1,48 +1,287 @@
 jobs:
-  build:
+  configure_env_vars:
+    <<: *defaults
+
+    steps:
+      - checkout
+
+      - run:
+          # Set TERMINUS_ENV and related environment variables.
+          # https://github.com/pantheon-systems/docker-build-tools-ci/blob/6.x/scripts/set-environment
+          name: setup-environment-vars
+          command: ./.ci/build/multidev-save
+
+      - run:
+          name: clone-bash-env
+          command: cp $BASH_ENV bash_env.txt
+
+      # https://circleci.com/docs/2.0/workflows/#using-workspaces-to-share-data-among-jobs
+      - persist_to_workspace:
+          root: .
+          paths:
+            - bash_env.txt
+
+  static_tests:
+    <<: *defaults
+
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+            - composer-cache-{{ checksum "composer.lock" }}
+            - composer-cache-
+
+      - run:
+          # Set TERMINUS_ENV and related environment variables.
+          # https://github.com/pantheon-systems/docker-build-tools-ci/blob/6.x/scripts/set-environment
+          name: setup-environment-vars
+          command: ./.ci/build/multidev-save
+
+      - run:
+          name: run static tests
+          command: ./.ci/test/static/run
+
+      - save_cache:
+          key: composer-cache-{{ checksum "composer.lock" }}
+          paths:
+            - $HOME/.composer/cache
+
+  build_php:
+    <<: *defaults
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+            - composer-cache-{{ checksum "composer.lock" }}
+            - composer-cache-
+            - vendor-{{ checksum "composer.lock" }}
+            - vendor-
+
+      - run:
+          # Set TERMINUS_ENV and related environment variables.
+          # https://github.com/pantheon-systems/docker-build-tools-ci/blob/6.x/scripts/set-environment
+          name: setup-environment-vars
+          command: ./.ci/build/multidev-save
+
+      - run:
+          name: run PHP build step
+          command: ./.ci/build/php
+
+      - save_cache:
+          key: vendor-{{ checksum "composer.lock" }}
+          paths:
+            - $CIRCLE_WORKING_DIRECTORY/vendor
+
+      - persist_to_workspace:
+          root: .
+          paths:
+            - web
+            - vendor
+            - drush
+
+  deploy_to_pantheon:
+    <<: *defaults
+    steps:
+      - checkout
+
+      # Attach the workspace
+      - attach_workspace:
+          at: /tmp
+
+      - run:
+          # Set TERMINUS_ENV and related environment variables.
+          # https://github.com/pantheon-systems/docker-build-tools-ci/blob/6.x/scripts/set-environment
+          name: setup-environment-vars
+          command: ./.ci/build/multidev-save
+
+      # Rsync the build artifact pieces web directory
+      - run:
+          name: sync build artifact
+          command: rsync -rz /tmp/web /tmp/vendor /tmp/drush .
+
+      # Deploy to Pantheon
+      - run:
+          name: deploy to Pantheon
+          command: ./.ci/deploy/pantheon/dev-multidev
```

Comment:
This file isn't included in this commit, but it's worth mentioning that I just ran into confusing deployment problems because in it, the Drupal cache is cleared only _after_ the db is updated and configuration imported. When I had [this problem with config_split](https://github.com/thinkshout/drupal-project/pull/92/files#r726957854), even after I got the settings files choosing the right config split for a Pantheon environment, the old config split was somehow cached as being the current one. Long story short, I think that cache needs to be cleared (at least once) before `updb` and `cim`.

Reply:
It doesn't come up often -- I didn't even realize they didn't do this already. Since the `./.ci/deploy/pantheon/dev-multidev` file gets committed, we could tell people to add a line clearing the cache before this one: https://github.com/pantheon-systems/example-drops-8-composer/blob/master/.ci/deploy/pantheon/dev-multidev#L27, but since I haven't run into the same issue using these tools, I hesitate to do it.